sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -559,7 +567,6 @@ case class JsonToStructs(
       (rows: Iterator[InternalRow]) => if (rows.hasNext) rows.next().getMap(0) else null
   }
 
-  val nameOfCorruptRecord = SQLConf.get.getConf(SQLConf.COLUMN_NAME_OF_CORRUPT_RECORD)
```

Comment:
`nameOfCorruptRecord` is used only in `parser` below. I would try to move it there:
```scala
@transient lazy val parser = {
  val parsedOptions = new JSONOptions(
    options,
    timeZoneId.get,
    SQLConf.get.getConf(SQLConf.COLUMN_NAME_OF_CORRUPT_RECORD))
```